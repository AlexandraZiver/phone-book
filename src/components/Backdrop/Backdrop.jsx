import PropTypes from "prop-types";
import React from "react";

import style from "./Backdrop.module.scss";

const Backdrop = ({ children }) => {
  return <div className={style.Container}>{children}</div>;
};

Backdrop.propTypes = {
  children: PropTypes.node,
};

export default Backdrop;
