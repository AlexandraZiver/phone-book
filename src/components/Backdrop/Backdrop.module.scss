@import "../../styles/mixins/header.scss";

.Container {
  font-size: 18px;
  text-align: center;
  height: 100vh;
  @include header();
}
