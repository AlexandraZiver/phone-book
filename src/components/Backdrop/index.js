export { default as Backdrop } from "./Backdrop.jsx";
