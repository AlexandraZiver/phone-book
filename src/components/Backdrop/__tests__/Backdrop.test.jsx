import { screen, render } from "@testing-library/react";
import Backdrop from "../Backdrop";

describe("Backdrop render testing", () => {
  const { container } = render(
    <Backdrop>
      <div>Hello</div>
    </Backdrop>,
  );

  const backdropElement = container.querySelector(".Container");
  const childrenElement = screen.getByText("Hello");

  test("testing Backdrop and ChildrenElement", () => {
    expect(backdropElement).toBeInTheDocument();
    expect(childrenElement).toBeInTheDocument();
    expect(backdropElement).toHaveStyle({ fontSize: "18px;" });
    expect(backdropElement).toMatchSnapshot();
  });
});
