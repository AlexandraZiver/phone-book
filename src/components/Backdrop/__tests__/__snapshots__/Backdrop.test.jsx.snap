// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backdrop render testing testing Backdrop and ChildrenElement 1`] = `
<div
  class="Container"
>
  <div>
    Hello
  </div>
</div>
`;
